feat: Add Realm Database

PruthiviRaj27 opened the pull request:
- In this commit, we added the Realm database via the swift package manager.
- Added `OfflineAsset` model and Initialize Realm database in `TPStreamsSDK`.
- Added Generic ObjectManager class for add, get object form database.
- Added Realm database dependencies in `Package.swift` and `TPStreamsSDK.podspec` files.

PruthiviRaj27 committed:
add database

PruthiviRaj27 committed:
refcator

PruthiviRaj27 committed:
refcator

Karthik-0 commented on Source/Database/Model/OfflineAsset.swift:
```diff
@@ -0,0 +1,45 @@
+}
+
+extension OfflineAsset {
+    func update(_ attributes: [String: Any]) {
+        let realm = try! Realm()
+        try! realm.write {
+            for (key, value) in attributes {
+                self[key] = value
+            }
+        }
+    }
```
Can this be extracted

Karthik-0 commented on Source/Database/ObjectManager.swift:
```diff
@@ -0,0 +1,23 @@
+       }
+    }
+
+    func get(assetId: Any) -> T? {
```
```suggestion
    func get(id: Any) -> T? {
```

Karthik-0 commented on Source/TPStreamsSDK.swift:
```diff
@@ -50,6 +52,11 @@ public class TPStreamsSDK {
     }
+    
+    private static func initializeDatabase() {
+        let config = Realm.Configuration(schemaVersion: 1)
+        Realm.Configuration.defaultConfiguration = config
```
```suggestion
        Realm.Configuration.defaultConfiguration = Realm.Configuration(schemaVersion: 1)

```

PruthiviRaj27 committed:
refcator

PruthiviRaj27 committed:
feat: Add Non-DRM video download support (#51)

- In this commit, we added Offline download support for Non-DRM videos via API
- Added Test case for add and get object operations

PruthiviRaj27 committed:
feat: Add download list (#57)

- In this commit, we added a download list to show the downloaded videos in Swift UI
- Added pause and resume options to the downloads

PruthiviRaj27 committed:
feat: Add `TPStreamsDownloadDelegate` (#59)

- In this commit, we added `TPStreamsDownloadDelegate` to monitor the download state changes we included methods for download start, pause, resume, complete

PruthiviRaj27 committed:
feat: Add support play downloaded non drm video (#60)

- In this commit, we added support to play downloaded video
- Added a new constructor with `OfflineAsset` param in `TPAVPlayer`

PruthiviRaj27 committed:
feat: Add seperate `OfflineAsset` model (#62)

- In this commit, we replace `OfflineAsset` with `OfflineAssetEntity` and made `OfflineAssetEntity` as internal
- We added the `OfflineAsset` model from external use
